backend/czi_hosted/data_common/matrix_loader.py

```diff
@@ -217,19 +12,28 @@ class MatrixDataType(Enum):
 
 
 class MatrixDataLoader(object):
-    def __init__(self, location, matrix_data_type=None, app_config=None):
-        """ location can be a string or DataLocator """
+    def __init__(self, location, app_config=None, url_dataroot=None, matrix_data_type=None):
```

Comment:
this routine would benefit from a bit more error checking, eg, throw on missing dataset_config.  It is cheap, and helps enforce assumptions downstream.